src/maker/config.rs

```diff
@@ -193,6 +194,17 @@ impl MakerConfig {
             .unwrap_or(default_config.connection_type),
         })
     }
+
+    // Method to write the provided TOML string to a file
+    pub fn write_to_file(self, path: &PathBuf, toml_data: String) -> std::io::Result<()> {
+        if let Some(parent) = path.parent() {
+            std::fs::create_dir_all(parent)?;
+        }
+        let mut file = std::fs::File::create(path)?;
+        file.write_all(toml_data.as_bytes())?;
+        file.flush()?;
+        Ok(())
+    }
```

Comment:
Since writing to a file is now handled by a method, the toml_data string can be created within the method itself, rather than being generated at the call site and passed in.